src/components/developers/apiLogs/ApiLogs.tsx

```diff
@@ -0,0 +1,167 @@
+import { Button, Typography } from 'lago-design-system'
+import { useCallback, useEffect, useLayoutEffect, useMemo, useRef } from 'react'
+import { generatePath, useNavigate, useParams, useSearchParams } from 'react-router-dom'
+
+import {
+  ApiLogsAvailableFilters,
+  Filters,
+  formatFiltersForApiLogsQuery,
+} from '~/components/designSystem/Filters'
+import { ApiLogDetails } from '~/components/developers/apiLogs/ApiLogDetails'
+import { ApiLogsTable } from '~/components/developers/apiLogs/ApiLogsTable'
+import { API_LOG_ROUTE } from '~/components/developers/DevtoolsRouter'
+import { ListSectionRef, LogsLayout } from '~/components/developers/LogsLayout'
+import { API_LOGS_FILTER_PREFIX } from '~/core/constants/filters'
+import { getCurrentBreakpoint } from '~/core/utils/getCurrentBreakpoint'
+import { ApiLogItemFragment, useGetApiLogsQuery } from '~/generated/graphql'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import { useDeveloperTool } from '~/hooks/useDeveloperTool'
+
+gql`
+  fragment ApiLogItem on ApiLog {
+    requestId
+    httpMethod
+    httpStatus
+    requestPath
+    loggedAt
+  }
+
+  query getApiLogs(
+    $page: Int
+    $limit: Int
+    $requestIds: [String!]
+    $fromDate: ISO8601Date
+    $toDate: ISO8601Date
+    $apiKeyIds: [String!]
+    $httpMethods: [HttpMethodEnum!]
+    $httpStatuses: [HttpStatus!]
+    $requestPaths: [String!]
+  ) {
+    apiLogs(
+      page: $page
+      limit: $limit
+      fromDate: $fromDate
+      toDate: $toDate
+      apiKeyIds: $apiKeyIds
+      httpMethods: $httpMethods
+      httpStatuses: $httpStatuses
+      requestIds: $requestIds
+      requestPaths: $requestPaths
+    ) {
+      collection {
+        ...ApiLogItem
+      }
+      metadata {
+        currentPage
+        totalPages
+      }
+    }
+  }
+`
+
+export const ApiLogs = () => {
+  const { translate } = useInternationalization()
+  const navigate = useNavigate()
+  const { logId } = useParams<{ logId: string }>()
+  const [searchParams] = useSearchParams()
+  const { size } = useDeveloperTool()
+  const logListRef = useRef<ListSectionRef>(null)
+
+  const filtersForApiLogsQuery = useMemo(() => {
+    return formatFiltersForApiLogsQuery(searchParams)
+  }, [searchParams])
+
+  const getApiLogsResult = useGetApiLogsQuery({
+    variables: { limit: 20, ...filtersForApiLogsQuery },
+    notifyOnNetworkStatusChange: true,
+  })
+
+  const { data, loading, refetch } = getApiLogsResult
+
+  const navigateToFirstLog = useCallback(
+    (logCollection?: ApiLogItemFragment[], currentSearchParams?: URLSearchParams) => {
+      if (logCollection?.length) {
+        const firstLog = logCollection[0]
+
+        if (firstLog && getCurrentBreakpoint() !== 'sm') {
+          const path = generatePath(API_LOG_ROUTE, { logId: firstLog.requestId })
+          const query = currentSearchParams?.toString()
+          const search = query ? `?${query}` : ''
+
+          navigate(`${path}${search}`, {
```

Comment:
You can use the `search` attribute in the navigate options.
Then remove the logic handling the `?` 
Something like this should work
```
navigate(
    generatePath(API_LOG_ROUTE, { logId: firstLog.requestId }), 
    { 
        search: currentSearchParams?.toString()
        replace: true
     }
 )
 ```